### DISCUSSION TITLE
Правила кода в демках

### DISCUSSION THREAD

**Discussion by furtivite** [doka-guide/content#912] [Разное]
Кажется, стоит прийти к единым стилям во всех частях контента и на платформе, например, если договорились использовать БЭМ, то использовать его везде. 

У нас в статьях есть демки, и если они достаточно сложные и состоят из нескольких блочных элементах, хорошо бы использовать в них БЭМ. Плюс описать это правило в документации для потенциальных внешних авторов. 

Что думаете?

**Comment by skorobaeus** [comment]
Если коротко, то "надо, так надо, но хотелось бы, чтобы было не надо". Если подробнее, то вот:

1. Фрагмент кода из демки почти всегда есть в тексте самой статьи ([пример](https://doka.guide/css/before/)). С БЭМ классами такие фрагменты станут более громоздкими и трудными для восприятия. Делать по-разному — в статье без БЭМ, в демке с БЭМ — нехорошо.

2. Неясен критерий достаточной сложности. В [этом демо](https://doka.guide/css/after/demos/word.html) 4 блочных элемента, но само демо простое.

3. Дока не про БЭМ...

4. Небольшое наблюдение: из 180 демок, которые я оформила на сегодняшний день, БЭМ изначальным автором был использован, наверное, в 5%-10% случаев.

**Comment by nlopin** [comment]
Я, как полный профан в вёрстке, не имею позиции про БЭМ. Со стороны кажется, что обсуждать правильный БЭМ можно бесконечно. Есть ли способ как-то это проверять автоматически, если мы договоримся его использовать везде? 

По джаваскрипту предлагаю просто запускать prettier в значениях по умолчанию, чтобы просто соблюсти единый стиль кода
